### DISCUSSION TITLE
Jumlah outliers

### DISCUSSION THREAD

**Discussion by Fizkal72** [teamalgoritma/community#167] [Q&A] [answered]
Ada gak cara utk menghitung jumlah outliers secara otomatis?

**Comment by nisa-basalamah** [comment, marked as answer]
Halo @Fizkal72,

Untuk mengetahui jumlah outliers pada suatu data, bapak dapat menggunakan 2 cara sebagai berikut:

**Cara 1:** Metode statistik dengan memanfaatkan nilai quartile dan IQR. Berikut penjelasan untuk langkahnya,
1. Menghitung nilai 1st quartile, `Q1= df.quantile(0.25) ` 
2. Menghitung nilai 3rd quartile, `Q3= df.quantile(0.75)`
3. Menghitung nilai IQR (interquartile range), `IQR= Q3-Q1`
4. Melakukan subsetting terhadap data (df) untuk mendapatkan data outliers, `outliers_data = df[( (df<(Q1-1.5*IQR)) | (df>(Q3+1.5*IQR)) )]`
5. Mendapatkan jumlah data outliers dengan menggunakan fungsi berikut, `len(outliers_data)`

> **Notes:** `.quartile()` merupakan fungsi yang dapat digunakan untuk menghitung nilai quartile. Nilai parameter 0.25 digunakan pada saat mencari nilai Q1, 0.5 untuk Q2, dan 0.75 untuk Q3.

**Cara 2:** Menggunakan method `boxplot_stats` dari `matplotlib.cbook`. Dengan menerapkan method `boxplot_stats`, bapak dapat memperoleh beberapa nilai statistik dari boxplot seperti: mean (rata-rata), IQR (interquartile range), Q1, Q2, Q3, serta **fliers** yang merupakan list berupa nilai outliers.

> **Notes:** Untuk cara penerapan lengkapnya, bapak dapat membaca dan mempraktekkan langkah-langkah yang dijelaskan pada referensi berikut: [Bagaimana cara mengekstrak nilai statistik dari boxplot?](https://askalgo-py.netlify.app/faq/dwv#bagaimana-cara-mengekstrak-nilai-statistik-dari-boxplot)

Semoga bermanfaat :)
